File: myblog-server/src/main/java/com/site/blog/model/vo/BlogDetailVO.java
package com.site.blog.model.vo;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.site.blog.model.entity.Category;
import com.site.blog.model.entity.Tag;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class BlogDetailVO {
    private String blogId;

    private String blogTitle;

    private Category blogCategory;

    private Long commentCount;

    private String blogCategoryIcon;
    private String preface;

    private String blogCoverImage;

    private Long blogViews;

    private List<Tag> blogTags;
    private String blogContent;

    private Boolean enableComment;
    private Boolean show;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private  LocalDateTime createTime;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;
    private Boolean deleted;
}

File: myblog-kt/src/main/kotlin/com/site/blog/model/vo/BlogDetailVO.kt
package com.site.blog.model.vo

import com.fasterxml.jackson.annotation.JsonFormat
import com.site.blog.model.entity.Category
import com.site.blog.model.entity.Tag
import java.time.LocalDateTime

data class BlogDetailVO  (
    var  blogId: String? = null,
    var  blogTitle: String? = null,
    var  blogCategory: Category? = null,
    var  commentCount: Long? = null,
    var  blogCategoryIcon: String? = null,
    var  preface: String? = null,
    var  blogCoverImage: String? = null,
    var  blogViews: Long? = null,
    var  blogTags: List<Tag?> = emptyList(),
    var  blogContent: String? = null,
    var  enableComment: Boolean? = null,
    var  show: Boolean? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    var  createTime: LocalDateTime? = null,

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    var  updateTime: LocalDateTime? = null,
    var  deleted: Boolean? = null,
)